docs: Added a new intro page for configuring external IdPs. (#7595)
zitadel/zitadel: docs/docs/guides/integrate/identity-providers/introduction.md

- **Automatic creation**: When enabled, this allows ZITADEL to automatically create a new user account if someone logs in with their external IdP credentials and no corresponding account exists in ZITADEL. This facilitates a smooth user onboarding experience by eliminating the need for manual account creation.

- **Automatic update**: This feature, when activated, allows ZITADEL to automatically update a user's profile information whenever changes are detected in the user's account on the external IdP. For example, if a user changes their last name in their Google or Microsoft account, ZITADEL will reflect this update in the user's account upon their next login.

- **Account creation allowed**: Determines whether new user accounts can be created in ZITADEL through the external IdP authentication process. Enabling this setting is crucial for allowing users who are new to your application to register and create accounts seamlessly via their existing external IdP accounts.

> - **Account linking allowed**: Enables existing ZITADEL accounts to be linked with identities from external IdPs. It requires that a linkable ZITADEL account already exists for the user attempting to log in with an external IdP. Account linking is beneficial for users who wish to associate multiple login methods with their ZITADEL account, providing flexibility and convenience in how they access your application.

Reviewer: should be auto update

## Where ZITADEL fits in

ZITADEL positions itself as the central hub or the identity broker in the interaction between your application and various external IdPs. It handles the orchestration of authentication requests and manages the seamless flow of identity verification between your application and the chosen IdP. This is achieved through a process known as federation, where ZITADEL acts as a federated identity provider, integrating IdPs via authentication protocols like OpenID Connect and SAML.

## Adding external identity providers to your application

With ZITADEL, you can enhance your application's accessibility by integrating social login IdPs such as Google or other IdPs such as EntraID (formerly known as AzureAD) and Auth0. For organizations with bespoke identity solutions, ZITADEL supports integration with custom-built IdPs that adhere to OpenID Connect or SAML protocols. By default, ZITADEL can serve as the primary user store for your applications. This centralizes user management and simplifies the authentication process, allowing users to sign in with their email and password, or via the external IdPs you've integrated.

> ## The advantages of using ZITADEL

Reviewer: We should also add something about the easy of use in b2b scenarios where you can have different idps for different customers

Author: Done.

The benefits of integrating ZITADEL for managing external IdPs are multiple:

- **No need for custom authentication code**: Your application only needs to interact with ZITADEL, which handles all communications with external IdPs. This abstraction saves you from the complexity of direct integration with multiple IdPs.

- **Unified protocol handling**: ZITADEL abstracts away the specific protocols used by different IdPs. Your application communicates with ZITADEL using a standard protocol (e.g., OpenID Connect), while ZITADEL takes care of the rest.

- **Centralized user management**: All user profiles are managed within ZITADEL, allowing for a unified view of user identities regardless of the IdP used for authentication.

- **Dynamic profile synchronization**: When users update their profiles on an external IdP, those changes are reflected in ZITADEL at the next login, ensuring that user data remains current.

- **Simplified account linking**: ZITADEL can link identities from multiple IdPs to a single user profile, facilitating a cohesive user experience across different authentication methods.

> - **Enhanced group and policy management**: With support for group synchronization and detailed policy configurations, ZITADEL offers granular control over user access and authentication processes.

Reviewer: What do you mean with that?
ZITADEL does not allow out of the box group and policy sync.
With actions it would be possible.

Author: I removed it altogether.

## The user journey 

- **Integration with external identity providers**: ZITADEL supports integrating a variety of external identity providers, including social logins like Google or GitHub, as well as custom IdPs that use OpenID Connect or SAML protocols.

- **Initiation of sign-in process**: Users select to sign in with an external IdP within your application.

- **Redirection to ZITADEL**: The application redirects the user to ZITADEL, which guides them to their chosen external IdP for authentication.

- **User authentication**: After successfully logging in at the external IdP site, users are redirected back to the application through ZITADEL, bringing along authentication tokens and profile information.

- **Account linking for new users**: If the user's identity from the external IdP does not exist in ZITADEL, they're presented with two options before moving forward:
  - **Create a new account**: Choosing this option creates a new ZITADEL account linked to the external IdP.

>   - **Autolinking with an existing local account**: Users have the option to link their new external identity to an existing local account in ZITADEL, enabling future logins with either their local account or the external IdP.

Reviewer: We do not provide autolinking, they need to link manually by authenticating the user to whom they want to link

Author: Done.

- **Profile pre-filling from external IdP**: ZITADEL uses information from the external IdP to pre-fill the user's profile, simplifying the account creation or linking process.

- **User option to update profile**: Users can review and, if necessary, update their pre-filled profile information.

- **Session creation and access granting**: After the account is created or linked and the profile is set, the application grants access by creating a session for the user based on their authenticated identity.

## Setting up external identity providers in ZITADEL

In ZITADEL, you have the flexibility to link an external Identity Provider (IdP) to your entire instance, making it the default option for all organizations within your instance, or to connect it exclusively to a specific organization. This setup allows organization members to leverage similar capabilities in self-service if permitted.

### Adjusting the custom login policy

The login policy can be set as a default at the instance level and can be customized for each organization. The configuration process varies slightly depending on your focus:

> - **For instance-wide settings**, navigate to: `$YOUR-DOMAIN/ui/console/instance?id=general`

Reviewer: I would name this default settings, as we are not really using instance settings anymore

Once in the settings:
- Access the **Login Behavior and Security** section to modify your login policy. Here, ensure you enable the option for **External IDP Allowed**.

![Allow External IDP](/img/guides/zitadel_allow_external_idp.png)

### Configuring IdP Providers

Access the settings page of your instance or the specific organization and select **Identity Providers**.

The ZITADEL Console will display a list of all the IdPs you've configured, along with available provider templates. Selecting any listed IdP will guide you through the process of configuring that specific Identity Provider.

![Identity Provider Overview](/img/guides/zitadel_identity_provider_overview.png)

## Available templates

In the guides below, you'll learn how to configure and set up your preferred external Identity Provider (IdP) in ZITADEL.

- [Google](./google)
- [Entra ID (OIDC)](./azure-ad-oidc)
- [Entra ID SAML](./azure-ad-saml)
- [GitHub](./github)
- [GitLab](./gitlab)
- [Apple](./apple)
- [LDAP](./ldap) 
- [Local OpenLDAP](./openldap.mdx)
- [OKTA generic OIDC](./okta-oidc)
- [OKTA SAML](./okta-saml)
- [Keycloak generic OIDC](./keycloak)
- [MockSAML](./mocksaml)

> - [JWT IdP](./jwt_idp)

Reviewer: These are not all available templates, some of them are based on the generic tempaltes. We just provide guides on how to configure them

Author: Changed.

- **Automatic creation**: When enabled, this allows ZITADEL to automatically create a new user account if someone logs in with their external IdP credentials and no corresponding account exists in ZITADEL. This facilitates a smooth user onboarding experience by eliminating the need for manual account creation.

- **Automatic update**: This feature, when activated, allows ZITADEL to automatically update a user's profile information whenever changes are detected in the user's account on the external IdP. For example, if a user changes their last name in their Google or Microsoft account, ZITADEL will reflect this update in the user's account upon their next login.

- **Account creation allowed**: Determines whether new user accounts can be created in ZITADEL through the external IdP authentication process. Enabling this setting is crucial for allowing users who are new to your application to register and create accounts seamlessly via their existing external IdP accounts.

> - **Account linking allowed**: Enables existing ZITADEL accounts to be linked with identities from external IdPs. It requires that a linkable ZITADEL account already exists for the user attempting to log in with an external IdP. Account linking is beneficial for users who wish to associate multiple login methods with their ZITADEL account, providing flexibility and convenience in how they access your application.

Reviewer: It might be worth putting this in a separat file and import here.
We could then also include that in all the identity provider guides. We already have that there, but with a shorter description.
Would be nice to have it only once.

Author: Fabi, I thought about this and I think this longer description is suitable for the intro page and we can keep it short for the separate pages which include the text in one place anyway.

> ## Configure external IdPs at the organization level or at the instance level

Reviewer: ```suggestion
## Configure external IdPs at the organization level or on the default settings
```

> Deciding whether to configure an external Identity Provider (IdP) at the organization level or the instance level in ZITADEL depends on the scope of access and management you intend to provide. Here’s when to choose each option:

Reviewer: ```suggestion
Deciding whether to configure an external Identity Provider (IdP) at the organization level or in the default settings in ZITADEL depends on the scope of access and management you intend to provide. Here’s when to choose each option:
```

### Create an external IdP on an organization

- **Targeted access control**: When you want to allow authentication through the external IdP specifically for users of a particular organization. This is useful for businesses that manage multiple organizations within ZITADEL and require distinct authentication strategies for each.

- **Customized authentication flow**: If different organizations have unique requirements or policies around user authentication, configuring an IdP at the organization level allows for customized authentication flows that cater to the specific needs of each organization.

- **Delegated administration**: Enabling organizations to manage their own IdPs empowers them to administer their authentication mechanisms independently. This is beneficial in scenarios where organizations have the autonomy to choose their preferred IdPs or when they manage their user base directly.

> ### Create an external IdP on an instance

Reviewer: ```suggestion
### Create an external IdP as default
```

> - **Unified authentication strategy**: When you aim to provide a consistent authentication experience across all organizations within your ZITADEL instance. Configuring an IdP at the instance level applies the same authentication mechanism universally, simplifying management and user experience.

Reviewer: ```suggestion
- **Unified authentication strategy**: When you aim to provide a consistent authentication experience across all organizations within your ZITADEL instance. Configuring an IdP in the default settings applies the same authentication mechanism universally, simplifying management and user experience.
```

> - **Centralized management**: Setting up an IdP at the instance level is ideal for scenarios where a single administrative body oversees user authentication across all organizations. This approach centralizes the management of external IdPs, making it easier to maintain and update authentication policies.

Reviewer: ```suggestion
- **Centralized management**: Setting up an IdP in the default settings is ideal for scenarios where a single administrative body oversees user authentication across all organizations. This approach centralizes the management of external IdPs, making it easier to maintain and update authentication policies.
```

> - **Broad access needs**: If all users, regardless of their organization, require access to external IdPs for authentication, configuring the IdP at the instance level ensures that these options are available universally. This is particularly useful for platforms that serve a wide range of users with common access requirements.

Reviewer: ```suggestion
- **Broad access needs**: If all users, regardless of their organization, require access to external IdPs for authentication, configuring the IdP in the default settings ensures that these options are available universally. This is particularly useful for platforms that serve a wide range of users with common access requirements.
```
